src/Yaapii.Xambly/Directive/XpathDirective.cs

```diff
@@ -46,34 +47,44 @@ public class XpathDirective : IDirective
         /// </summary>
         private const string ABSOLUTE_XPATH_REGEX = @"^((?:\/(?!\/)).*)$";
 
-        /// <summary>
-        /// XPath factory.
-        /// </summary>
-        private readonly IArg _expr;
+        private readonly IArg expr;
+        private readonly string rootDefNamespacePrefix;
+        private readonly IDictionary<string, string> defaultNamespaceAndPrefix;
 
-        // <summary>
-        // Pattern to match root-only XPath queries.
-        // </summary>
-        //private static readonly Regex ROOT_ONLY = new Regex(@"/([^\/\(\[\{:]+)");
+        /// <summary>
+        /// XPATH directive.
+        /// Moves cursor to the nodes found by XPath.
+        /// </summary>
+        /// <param name="path">XPath</param>
+        /// <param name="rootDefNamespacePrefix">Default namespace prefix for use in XPath or empty string</param>

```

Comment:
isn't is simply the namespace for the xpath? That name seems overcomplicated.

Reply:
It is still too complicated for a variable name and usage. If it is really that complex, there should be objects which help using and understanding it in a simple way.

Reply:
Thanks, then we need extra objects I think